unable to compile postgres_scanner on gcc

### What happens?

unable to compile postgress_scanner on gcc 11.4

errors in postgress_scanner.cpp lines 205 and 234

postgres_scanner.cpp: In function ‘duckdb::unique_ptr<duckdb::GlobalTableFunctionState, std::default_delete<duckdb::GlobalTableFunctionState>, true> duckdb::PostgresInitGlobalState(duckdb::ClientContext&, duckdb::TableFunctionInitInput&)’:


postgres_scanner.cpp:205:16: error: could not convert ‘result’ from ‘unique_ptr<duckdb::PostgresGlobalState,default_delete<duckdb::PostgresGlobalState>,[...]>’ to ‘unique_ptr<duckdb::GlobalTableFunctionState,default_delete<duckdb::GlobalTableFunctionState>,[...]>’
  205 |         return result;
      |                ^~~~~~
      |                |
      |                unique_ptr<duckdb::PostgresGlobalState,default_delete<duckdb::PostgresGlobalState>,[...]>


postgres_scanner.cpp: In function ‘duckdb::unique_ptr<duckdb::LocalTableFunctionState, std::default_delete<duckdb::LocalTableFunctionState>, true> duckdb::GetLocalState(duckdb::ClientContext&, duckdb::TableFunctionInitInput&, duckdb::PostgresGlobalState&)’:


postgres_scanner.cpp:234:24: error: could not convert ‘local_state’ from ‘unique_ptr<duckdb::PostgresLocalState,default_delete<duckdb::PostgresLocalState>,[...]>’ to ‘unique_ptr<duckdb::LocalTableFunctionState,default_delete<duckdb::LocalTableFunctionState>,[...]>’
  234 |                 return local_state;
      |                        ^~~~~~~~~~~
      |                        |
      |                        unique_ptr<duckdb::PostgresLocalState,default_delete<duckdb::PostgresLocalState>,[...]>




### To Reproduce

make build details and gcc

cmake --build build/release --config Release
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found Git: /usr/bin/git (found version "2.34.1") 
-- git hash 401c8061c6, version v0.9.1


-- Extensions linked into DuckDB: [tpch, tpcds, parquet, jemalloc]
-- Extensions built but not linked: [postgres_scanner]
-- Tests loaded for extensions: [postgres_scanner]



### OS:

ubuntu 22.04

### PostgreSQL Version:

15.2   (Ubuntu 15.2-1.pgdg22.04+1)

### DuckDB Version:

0.9.1

### DuckDB Client:

CLI on unix or C++ lib.

### Full Name:

James Hackett

### Affiliation:

none

### Have you tried this on the latest `main` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

--- a/src/postgres_scanner.cpp
+++ b/src/postgres_scanner.cpp
@@ -202,7 +202,7 @@ static unique_ptr<GlobalTableFunctionState> PostgresInitGlobalState(ClientContex
 		result->collection = std::move(materialized);
 		result->collection->InitializeScan(result->scan_state);
 	}
-	return result;
+	return std::move(result);
 }
 
 static bool PostgresParallelStateNext(ClientContext &context, const FunctionData *bind_data_p,
@@ -231,7 +231,7 @@ static unique_ptr<LocalTableFunctionState> GetLocalState(ClientContext &context,
 
 	auto local_state = make_uniq<PostgresLocalState>();
 	if (gstate.collection) {
-		return local_state;
+		return std::move(local_state);
 	}
 	local_state->column_ids = input.column_ids;
 

